subsys/bluetooth/shell/has_client.c

```diff
@@ -127,6 +131,118 @@ static int cmd_has_client_read_presets(const struct shell *sh, size_t argc, char
 	return err;
 }
 
+static int cmd_has_client_preset_set(const struct shell *sh, size_t argc, char **argv)
+{
+	bool sync = false;
+	uint8_t index;
+	int err;
+
+	index = shell_strtoul(argv[1], 16, &err);
+	if (err < 0) {
```

Comment:
`shell_strtoul` does not set `err` in case of success, so `err` should be set to 0 before calling `shell_strtoul`

```suggestion
	int err = 0;

	index = shell_strtoul(argv[1], 16, &err);
	if (err < 0) {
```